src/network/network.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <memory>
+#include "common/assert.h"
+#include "common/logging/log.h"
+#include "enet/enet.h"
+#include "network/network.h"
+
+namespace Network {
+
+std::unique_ptr<RoomMember> g_room_member; ///< RoomMember (Client) for network games
+std::unique_ptr<Room> g_room;              ///< Room (Server) for network games
+
+bool Init() {
+    ASSERT_MSG(enet_initialize() == 0, "Error initalizing network");
+    g_room = std::make_unique<Room>();
+    g_room_member = std::make_unique<RoomMember>();
+    LOG_DEBUG(Network, "initialized OK");
+    return true;
+}
+
+Room* getRoom() {
```

Comment:
I don't think forwarding uinque_ptr is the right thing here either. Probably the best way is to forward the member functions of room and room_member instead of the members.

Alternatively I would suggest to use shared_ptr and then use
`shared_ptr<Room> GetRoom()`
and
`shared_ptr<RoomMember> GetRoomMember()`